### PR TITLE
test GetChildDetailsTask

### PR DESCRIPTION
**Pull request by Andati**

### CODE REVIEW

File: opensrp-child/build.gradle [hunk 1]
```diff
@@ -22,6 +22,7 @@ jacoco {
 android {
     compileSdkVersion androidCompileSdkVersion
     buildToolsVersion androidBuildToolsVersion
+    useLibrary 'org.apache.http.legacy'
```
Comment (ndegwamartin) [inline opensrp-child/build.gradle:25]: Why do we need this here?

File: opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+        Whitebox.setInternalState(getChildDetailsTask, "profilePhoto", imageView);
+        Whitebox.setInternalState(getChildDetailsTask, "initials", textView);
+
+        Resources resources = Mockito.mock(Resources.class);
+        when(baseActivity.getResources()).thenReturn(resources);
+
+        PowerMockito.mockStatic(DrishtiApplication.class);
+        PowerMockito.when(DrishtiApplication.getCachedImageLoaderInstance()).thenReturn(openSRPImageLoader);
+
```
Comment (ndegwamartin) [inline opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java:136]: We should verify it was INVISIBLE state first on this line then proceed with the invocation

Comment (ndegwamartin) [inline opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java:136]: We should verify it was INVISIBLE or GONE state first on this line then proceed with the invocation

File: opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        childDetails.put(Constants.KEY.HAS_PROFILE_IMAGE, "false");
+        CommonPersonObjectClient commonPersonObjectClient = new CommonPersonObjectClient("e34343-343434-67", new HashMap<String, String>(), "Roja");
+        commonPersonObjectClient.setColumnmaps(childDetails);
+
+        Whitebox.setInternalState(getChildDetailsTask, "profilePhoto", imageView);
+        Whitebox.setInternalState(getChildDetailsTask, "initials", textView);
+
+        Resources resources = Mockito.mock(Resources.class);
+        when(baseActivity.getResources()).thenReturn(resources);
+
```
Comment (ndegwamartin) [inline opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java:110]: We should verify it was in a VISIBLE state first on this line then proceed with the invocation

File: opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        commonPersonObjectClient.setColumnmaps(childDetails);
+
+        Whitebox.setInternalState(getChildDetailsTask, "initials", textView);
+
+        processChildNames.invoke(getChildDetailsTask, baseActivity, commonPersonObjectClient);
+        verify(textView).setVisibility(View.VISIBLE);
+    }
+
+    @Test
+    public void testUpdatePicture() throws Exception {
```
Comment (ndegwamartin) [inline opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java:93]: We can make the title here a bit more descriptive in the format -  **test[MethodName][ExpectedResult][Parameters]**  (Parameters optional if not present)

File: opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        getChildDetailsTask = new GetChildDetailsTask(baseActivity, "e34343-343434-67", view);
+    }
+
+    @Test
+    public void testProcessChildNames() throws Exception {
```
Comment (ndegwamartin) [inline opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java:58]: We can make the title here a bit more descriptive in the format - test[MethodName][ExpectedResult][Parameters] (Parameters optional if not present)

File: opensrp-child/build.gradle (hunk 1 above)
Comment (Andati) [reply opensrp-child/build.gradle:25]: the tests fail without this

File: opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java (hunk 3 above)
Comment (Andati) [reply opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java:110]: so, it's shown by default. Am just verifying that the setVisibility method is called

File: opensrp-child/build.gradle (hunk 1 above)
Comment (Andati) [reply opensrp-child/build.gradle:25]: some of the underlying logic is in core lib and core lib has this

File: opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java (hunk 3 above)
Comment (ndegwamartin) [reply opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java:110]: Intuitively this is true however we need the test to assert that this is indeed true before we invoke the message that alters that state

Comment (ndegwamartin) [reply opensrp-child/src/test/java/org/smartregister/child/task/GetChildDetailsTaskTest.java:110]: Actually verify is fine, my comment above would have held true if it was an _Assert_

**Pull request merged**